src/model/mod.rs

```diff
@@ -392,15 +397,328 @@ pub fn calculate_ratings(
+                    panic!("Head to head with team based match type is unsupported!")
+                }
+
+                if gwr.winners.len() > 1 || gwr.losers.len() > 1 {
+                    panic!("Head to head with more than 1 member per team is unsupported!");
+                }
+
+                // Set the winner to team red
+                if red_roster.len() == 0 {
+                    red_roster = gwr.winners.clone();
+                    blue_roster = gwr.losers.clone();
+                }
+
+                // Compare the scores based on id
+                if gwr.winners == red_roster {
+                    red_points += 1;
+                }
+
+                if gwr.winners == blue_roster {
+                    blue_points += 1;
+                }
+            }
+        }
+    }
+
+    red_roster = red_roster.into_iter().unique().collect();
+    blue_roster = blue_roster.into_iter().unique().collect();
+
+    let mut winner_team: Option<i32> = None;
+    let mut loser_team: Option<i32> = None;
+
+    if red_points > blue_points {
+        winner_team = Some(RED_TEAM_ID);
+        loser_team = Some(BLUE_TEAM_ID);
+    }
+
+    if blue_points > red_points {
+        winner_team = Some(BLUE_TEAM_ID);
+        loser_team = Some(RED_TEAM_ID);
+    }
+
+    if red_points == blue_points {
+        // The match is a tie
+        winner_team = None;
+        loser_team = None;
+
+        return MatchWinRecord {
+            match_id,
+            loser_roster: blue_roster,
+            winner_roster: red_roster,
+            winner_points: red_points,
+            loser_points: blue_points,
+            winner_team,
+            loser_team,
+            match_type: Some(match_type)
+        };
+    }
+
+    // Identify winning & losing rosters. If tie, default to red.
+    // In a head to head, the winning player is always red.
+
+    let winner_roster = if winner_team == Some(RED_TEAM_ID) {
+        red_roster.clone()
+    } else {
+        blue_roster.clone()
+    };
+
+    let loser_roster = if loser_team == Some(RED_TEAM_ID) {
+        red_roster
+    } else {
+        blue_roster
+    };
+
+    let winner_points = if winner_team == Some(RED_TEAM_ID) {
+        red_points
+    } else {
+        blue_points
+    };
+
+    let loser_points = if loser_team == Some(RED_TEAM_ID) {
+        red_points
+    } else {
+        blue_points
+    };
+
+    if match_type == MatchType::HeadToHead {
+        winner_team = Some(0);
+        loser_team = Some(0);
+    }
+
+    MatchWinRecord {
+        match_id,
+        loser_roster,
+        winner_roster,
+        winner_points,
+        loser_points,
+        winner_team,
+        loser_team,
+        match_type: Some(match_type)
+    }
+}
+
+/// For each player in the match, generate one [`PlayerMatchStats`] object.
+/// This allows us to identify how each player performed in the match.
+fn player_match_stats(matches: &[Match]) -> Vec<PlayerMatchStats> {
```

Comment:
Is this really need a separate function? I'm pretty sure you can calculate everything you need inside `calculate_processed_match_data` function

Reply:
For now, I'm doing it because that function is huge. I don't want to make it even more complex. We can optimize in the future.